controllers/workspace/provision/deployment.go

```diff
@@ -189,19 +217,29 @@ func getSpecDeployment(
 					Volumes:                       podAdditions.Volumes,
 					RestartPolicy:                 "Always",
 					TerminationGracePeriodSeconds: &terminationGracePeriod,
-					SecurityContext: &corev1.PodSecurityContext{
-						RunAsUser: user,
-						FSGroup:   user,
-					},
-					ServiceAccountName:           saName,
-					AutomountServiceAccountToken: nil,
+					SecurityContext:               GetDevWorkspaceSecurityContext(),
```

Comment:
Seems it lead Devworkspace to an endless reconcile loop.
The only way to catch the cause:
```
	if !cmp.Equal(specDeployment, clusterDeployment, deploymentDiffOpts) {
		clusterAPI.Logger.Info("deployment needs to be updated", "diff",  cmp.Diff(specDeployment, clusterDeployment, deploymentDiffOpts))
		clusterAPI.Logger.Info("Updating deployment...")
```
probably we need something like that on debug log level always...
So, 
```
&v1.Deployment{
...
- \t\t\t\tSecurityContext:       nil,
+ \t\t\t\tSecurityContext:       s\"&PodSecurityContext{SELinuxOptions:nil,RunAsUser:nil,RunAsNonRoot:nil,SupplementalGroups:[],FSGroup:nil,RunAsGroup:nil,Sysctls:[]Sysctl{},WindowsOptions:nil,FSGroupChangePolicy:nil,}\",
  \t\t\t\tImagePullSecrets:      nil,
  \t\t\t\tHostname:              \"\",
  \t\t\t\t... // 1 ignored and 13 identical fields
  \t\t\t},
  \t\t},
  \t\tStrategy:        {Type: \"Recreate\"},
  \t\tMinReadySeconds: 0,
  \t\t... // 2 ignored and 1 identical fields
  \t},
  \t... // 1 ignored field
  }
```

Reply:
Defaulting hits us again -- I'll fix it for the PR but this is an issue that should be resolved by moving to the `equality` package instead of `cmp`.

Reply:
Fixed for now.